frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -771,6 +771,60 @@ function ReaderFooter:unscheduleFooterAutoRefresh()
     UIManager:unschedule(self.autoRefreshFooter)
 end
 
+function ReaderFooter:toBeRepainted() -- xxx a better name here
+    local n = 1
+    local widget = UIManager:getTopWidget(n)
+--[[    while widget do
+        print("xxx n=", n, widget.name, widget.dimen)
+        for i,v in pairs(widget) do
+            print("xxx n-", i, v)
+        end
+        if widget.movable then
+            for i,v in pairs(widget.movable) do
+                print("xxx movable", i, v)
+            end
+        end
+        n = n + 1
+        widget = UIManager:getTopWidget(n)
+    end
+    n = 1
+    widget = UIManager:getTopWidget(n)
+]]
+
+    while widget do
+        if widget.name == "ReaderUI" then
+            return true
+--        elseif widget.name == "VirtualKeyboard" then
+--            return false
+            elseif widget.covers_fullscreen or widget.covers_footer == true then
+            -- (e.g. the virtual keyboard sets widget_covers_footer == true)
+            return false
+        elseif widget.ignore ~= "height" and widget.covers_footer ~= false then
+            -- widget.ignore == "height" is set in menu, widget.covers_footer is an optional hint for this here,
+            -- sample usage can be found in the AutoDim plugin.
+
+            -- This is just a simple check if any of the topmost widgets paint into the statusbar's y position.
+            -- Better would be to check if any topmost widget rectancle intersects with the footer content.
+            -- @todo: xxx
+            local widget_bottom_coordinate = 0 -- a widget may have no dimension (Hello TrapWidget)
+            if widget.movable then
+                widget_bottom_coordinate = widget.movable.dimen.y + widget.movable.dimen.h
```

Comment:
`dimen` is... weird.

You have *zero* guarantee that it:

* exists at all
* is actually a Geom and not just a plain table with possibly only a few of the four keys set
* is actually meaningful (a lot of stuff just chucks the Screen dimensions at it).
* is actually up-to-date (a lot of widgets only compute it during `paintTo`).

TL;DR: I'd be *extremely* wary of using `dimen` in any way, shape or form outside of UIManager's `_repaint` method.

Reply:
Case in point: ReaderFooter itself jumps through lot of shitty hoops to compute its own height *before* drawing...

https://github.com/koreader/koreader/blob/a23c30f862aea0cd553fa7491edfc68224e931be/frontend/apps/reader/modules/readerfooter.lua#L2196-L2223

Reply:
Thanks for your elaborations. In that case I will stop the work on this PR (before it get never ending :) )
Feel free to close it.